src/components/side-panel/side-panel.vue

```diff
@@ -0,0 +1,78 @@
+<script setup lang="ts">
+import { watch } from 'vue'
+import { storeToRefs } from 'pinia'
+
+import LayerPanel from '@/components/layer-panel/layer-panel.vue'
+import LegendsPanel from '@/components/legends/legends-panel.vue'
+import StylePanel from '@/components/style-selector/style-panel.vue'
+import MyMapsPanel from '@/components/my-maps/my-maps-panel.vue'
+import { screenSizeIsAtLeast } from '@/services/common/device.utils'
+import { useAppStore } from '@/stores/app.store'
+
+const appStore = useAppStore()
+const {
+  layersOpen,
+  legendsOpen,
+  myMapsOpen,
+  styleEditorOpen,
+  themeGridOpen,
```

Comment:
Thanks for the refactoring! Maybe we could even put all these into one `sidePanelOpen` variable with predefined values defining its state in the future (not within this PR). WDYT?

Reply:
Do you mean something like :
```
const sidePanelOpen = ref({
layersOpen: true,
legendsOpen: true,
myMapsOpen: true,
...
})
```

I am not a big fan of this solution because it means when a component is linked to themeGridOpen property only (for example), it will also trigger a refresh when the other properties are changed.

Reply:
Yep, good point. I was rather thinking of something like:
```
type SidePanelOpenState = 'layersOpen' | 'legendsOpen' | ...
const sidePanelOpenState = ref<SidePanelOpenState>('layersOpen')
```
or maybe using an enum, but just as an idea for future refactoring...

Reply:
Oh! right! this is much better than what I thought. Well, why not :). I am merging this PR, feel free to improve this side panel state.